schema-registry/src/main/java/io/streamnative/pulsar/handlers/kop/schemaregistry/model/impl/PulsarSchemaStorage.java

```diff
@@ -331,11 +331,15 @@ private void applyOpToLocalMemory(Op op) {
             try {
                 compatibility.put(op.subject, CompatibilityChecker.Mode.valueOf(op.compatibilityMode));
             } catch (IllegalArgumentException err) {
-                log.error("Unrecognized mode, skip op", op);
+                log.error("Unrecognized mode, skip op", err);
             }
         } else {
-            SchemaEntry schemaEntry = op.toSchemaEntry();
-            schemas.put(schemaEntry.id, schemaEntry);
+            if (op.status == SchemaStatus.DELETED) {
```

Comment:
Will the deleted schema delete on the topic?

Reply:
Yes. Deleting a subject will delete all the schemas that were registered into the subject. See https://docs.confluent.io/platform/current/schema-registry/schema-deletion-guidelines.html#soft-delete-a-schema

And currently KoP does not support deleting a specific version.